### ISSUE TITLE
FaunaAdapter appears to create distinct verification token from EmailProvider

### ISSUE THREAD

**Issue by tatwater** [nextauthjs/adapters#354]
### Title

FaunaAdapter appears to create distinct verification token from EmailProvider

### How to reproduce ☕️

In `[...nextauth].js`, force a known verification token:
    export default NextAuth({
      providers: [
        EmailProvider({
          server: {...},
          from: '...',
          async generateVerificationToken() {
            return 'ABC123'
          },
        }),
      ],
      adapter: FaunaAdapter(client),
      debug: true,
      ...
    });

Upon successful email send, the see the following log in terminal (note token is different):
    [next-auth][debug][adapter_createVerificationToken] {
      args: [
        {
          identifier: '...',
          token: 'b767433423cf774e0eea9db84e2a650c9286bbb8bd7a5656110a0eeec01ac1b3',
          expires: 2021-12-27T03:20:08.192Z
        }
      ]
    }

Check Fauna Dashboard online, Collections > verification_tokens to find new entry, matching debug log:
    {
      "ref": Ref(Collection("verification_tokens"), "319098200552112720"),
      "ts": 1640574608310000,
      "data": {
        "identifier": "...",
        "token": "b767433423cf774e0eea9db84e2a650c9286bbb8bd7a5656110a0eeec01ac1b3",
        "expires": Time("2021-12-27T03:20:08.192Z")
      }
    }

Check email, copy link for "Sign in" button and paste (note token is as intentionally set): `http://localhost:3000/api/auth/callback/email?callbackUrl=http%3A%2F%2Flocalhost%3A3000&token=ABC123&email=...`

Obviously receive "Unable to sign in" error page, confirm issue by checking debug log in terminal:
    [next-auth][debug][adapter_useVerificationToken] {
      args: [
        {
          identifier: 'teagan@sous.app',
          token: 'b767433423cf774e0eea9db84e2a650c9286bbb8bd7a5656110a0eeec01ac1b3'
        }
      ]
    }

### Your question/bug report 📓

By forcing my own, known verification token, I can demonstrate that what's being generated by the FaunaAdapter and stored in the database disregards what is either set manually or automatically generated by the EmailProvider. These tokens then obviously don't match, causing authentication to fail.

I have never contributed to an open source project before and I'm not sure how all the components within NextAuth work together, but I am willing to poke around and see what I can find that might be useful. If anyone can point me in the right direction that would be great!

### Contributing  🙌🏽

Yes, I am

**Comment by tatwater** [comment]
Should add:
`@next-auth/fauna-adapter` @ 1.0.1
`faunadb` @ 4.4.1
`next` @ 12.0.3
`next-auth` @ 4.0.6
`next-pwa` @ 5.4.4
`nodemailer` @ 6.7.0

**Comment by balazsorban44** [comment]
The token that is saved in the database is hashed so if the database is compromised xyou wouldn't be able to use that token to sign in on behalf of the users.

You should not care about the value saved i  the database, but rather the one that is sent in the email.

the email should contain the url with `token=your-token` and if you claim that's not the case, then indeed it seems to be a bug.

**Comment by tatwater** [comment]
That makes total sense, okay! In that case I must have not set something up properly since every attempt to sign up results in an "Unable to sign in" notice stating "The sign in link is no longer valid. It may have been used already or it may have expired" — has anyone had this experience or know where I could start looking for potential solutions? Thanks!

**Comment by balazsorban44** [comment]
We run tests on the adapters to have confidence in them working as intended. Please provide a reproduction. Much easier to help.

**Comment by tatwater** [comment]
Okay so I started a fresh project from scratch with just the basics to test this issue, and the problem remains. I've published the repo [here](https://github.com/tatwater/nextauth-fauna) for you or others to check out! Thanks for the help :)

**Comment by balazsorban44** [comment]
Your reproduction doesn't seem to use `generateVerificationToken` 🤔

**Comment by tatwater** [comment]
Yep! I left that out because now I know I can trust that the adapter & provider are using the same generated token by default, so I didn't think I needed to force a specific one. Is that not true?

**Comment by balazsorban44** [comment]
I'm slightly confused. I thought this issue was opened because the `generateVerificationToken` method did not work as expected.

I just tested it locally again, and the Fauna adapter works as expected, both with and without that method. :confused:

This is likely not a bug in `next-auth` or the adapter, converting to a discussion.
